app/src/main/java/com/ichi2/apisample/MainActivity.java

```diff
@@ -46,118 +50,317 @@
-            }
-        });
-        selectInterval.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
-            @Override
-            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-                clearAddedInputFilename();
-                refreshExisting();
-            }
-
+        textFilename.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onNothingSelected(AdapterView<?> adapterView) { }
-        });
-        seekTempo.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
-                TextView label = findViewById(R.id.labelTempoValue);
-                label.setText(Integer.toString(seekBar.getProgress()));
-                clearAddedInputFilename();
-                refreshExisting();
+            public void onClick(View view) {
+                if (filenames.length > 1) {
+                    StringBuilder msg = new StringBuilder();
+                    final String[][] orderedPermutationKeys = getOrderedPermutationKeys();
+                    for (int i = 0; i < filenames.length; i++) {
+                        if (msg.length() > 0) {
+                            msg.append(getString(R.string.filenames_list_separator));
+                        }
+                        if (i < orderedPermutationKeys.length) {
+                            final String startNote = orderedPermutationKeys[i][1] + orderedPermutationKeys[i][0];
+                            final String interval = orderedPermutationKeys[i][2];
+                            msg.append(getString(R.string.filenames_list_item_with_key, i + 1, filenames[i], startNote, interval));
+                        } else {
+                            msg.append(getString(R.string.filenames_list_item, i + 1, filenames[i]));
+                        }
+                    }
+                    new AlertDialog.Builder(MainActivity.this)
+                            .setMessage(msg)
+                            .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialogInterface, int i) {
+                                    dialogInterface.dismiss();
+                                }
+                            })
+                            .show();
+                }
             }
-
-            @Override public void onStartTrackingTouch(SeekBar seekBar) { }
-            @Override public void onStopTrackingTouch(SeekBar seekBar) { }
         });
+        labelNote.setOnLongClickListener(new OnFieldCheckLabelLongClickListener(checkNotes));
+        for (CheckBox checkNote : checkNotes) {
+            checkNote.setOnCheckedChangeListener(new OnFieldCheckChangeListener());
```

Comment:
It probably makes sense to use single "OnFieldCheckChangeListener" for all the controls instead of creation for each? (the same for other listeners)

Reply:
Depends on the logic that could potentially go into this class. But I'm not sure if this would be a good decision in a long run